crates/astria-sequencer-relayer/tests/blackbox/main.rs

```diff
@@ -51,14 +51,11 @@ async fn report_degraded_if_block_fetch_fails() {
         actual: readyz.json::<serde_json::Value>().await.unwrap(),
     );
 
+    // don't mount another block, so the relayer will fail to fetch the block
     let abci_guard = sequencer_relayer.mount_abci_response(1).await;
-    let block_guard = sequencer_relayer.mount_bad_block_response(1).await;
     timeout(
         Duration::from_millis(2 * sequencer_relayer.config.block_time),
-        futures::future::join(
-            abci_guard.wait_until_satisfied(),
-            block_guard.wait_until_satisfied(),
-        ),
+        abci_guard.wait_until_satisfied(),
```

Comment:
Thought about this some more: this actually is much more flaky than before.

This test now relies on the healthz request triggering after the block-request going through. That's because the timeout will already have returned after `abci_guard` is satisfied.

But there is actually no guarantee that the healthz is peformed after the block-request is done.

Reply:
@SuperFluffy i updated this to mount a bad block again, should be good now